### ISSUE TITLE
Documentation for generating Visual Studio Solution from qbs

### ISSUE THREAD

**Issue by shiyuge** [bjorn/tiled#1956]
I am a windows user and I would like to generate a Visual Studio Solution file.
I think there are many developers like me who prefer to use some other 3rd party IDE other than Qt Creator.

Normally, one could generate a vs sln by migrating the `.pro` file from a vs plugin, Qt VS Tools.
This plugin uses qmake. 
But in tiled.pro, I see
```
win* {
    message("Building Tiled for Windows using qmake is no longer supported")
    error("Use the tiled.qbs project file instead")
}
```
so it's impossible to do it this way.

Getting the error message, I looked up qbs online and run these commands:
```
qbs setup-qt --detect
qbs setup-toolchains --detect
qbs generate -g visualstudio2017
```

No error message occurred but I can't find a vs file anywhere. I get one `default.bg.lock` which I believe is some intermediate product in the qbs generate process. I think the process could be like this: the qbs generator locks current profile, then something goes wrong that breaks the generating process leaving only the lock.

I think the root course for my problem could be some qbs bugs, but I can't help but wonder:

1. Why is it that building Tiled for Windows using qmake is no longer supported? Can we restore it?
2. Would it be helpful if we have some additional documentation on how to generate a vs solution file? Many other programmers could have the same demand.

Thank you.

**Comment by bjorn** [comment]
> Why is it that building Tiled for Windows using qmake is no longer supported? Can we restore it?

It is time consuming to maintain multiple build systems, and qmake is more hacky and tricky to support. I don't remember what issue prompted me to put that message there, but in general my goal is to get rid of the qmake project files entirely (issue #1484). Of course, it could be restored somehow, but I think there are better ways to spend the time.

> Would it be helpful if we have some additional documentation on how to generate a vs solution file? Many other programmers could have the same demand.

As far as I know you're the first to try generating VS project files using Qbs for Tiled. I do not know how well this option is supported.

@jakepetroules, would you be able to help us getting that to work? Or is there another way in which the Qbs project files can be used in Visual Studio?

**Comment by bjorn** [comment]
Jake has let me know that he has no time to help out with this issue since he's no longer working on Qbs.

So, we have [the documentation](https://doc.qt.io/qbs/generators.html), and if there is a bug we can open an issue about it. However, when I try to execute `qbs generate -g visualstudio2017` on Linux I do get a `Tiled.sln` file and a lot of `vcxproj` and `vcxproj.filters` files, so it seems to do more than in your case @shiyuge.

I'm using Qbs version 1.11.1.

**Comment by shiyuge** [comment]
Looks like I did not set the tool-chain properly. I modified qbs profiles manually and now it works.

Do you think I can modify readme.md, adding some directions regarding how to build on windows with qbs and then make a pull request?

**Comment by Phlosioneer** [comment]
This still needs to be documented.

**Comment by eartheath** [comment]
> @shiyuge Hello, I have the same problem. Can you tell me how to generate valid VS solution and how you modified the qbs?
I execute `qbs generate -g visualstudio2017` in the root directory of `Tiled project` on Windows and get a `Tiled.sln` and a lot of related files, which are valid in `Visual Studio 2017` because I successfully ran the GUI of `Tiled` after compiling with `VS`. In fact, I found that `VS` also builds `Tiled project` through `QBS`. I hope I can help you.
